### DISCUSSION TITLE
Keep filtered records without re-applying filter

### DISCUSSION THREAD

**Discussion by demobiel** [filamentphp/filament#10583] [Help]
Hi,

is it possible to keep the "filtered" records if you change the value.

For example: you want to see all unpaid invoices, and perform some actions on it. One of those actions flags the invoice as being paid, but want to keep the invoice visible.

Kind regards

Andries

**Comment by leandrocfe** [comment]
[persistFiltersInSession](https://filamentphp.com/docs/3.x/tables/filters/getting-started#persist-filters-in-session) ?

**Comment by demobiel** [comment]
Hi @leandrocfe 
thanks for your reply. This function stores the filters in the session so they are reapplied when you back to the table. However this is not what I am looking for.

Ideally the filter would only be applied on a certain action of the user -> perform filter (or submit, ...), and the records that are "found" after applying the filter are kept in the table, even if their value would change so that it no longer meets the criteria of the filter.

For example: Show all invoices that are unpaid, changing the status to paid (via modal action, or bulkaction) would keep the record visible, now they disappear.
